src/services/codefixes/importFixes.ts

```diff
@@ -321,12 +351,52 @@ namespace ts.codefix {
             cache.set(result, host.getProjectVersion?.());
         }
         host.log?.(`getSymbolToExportInfoMap: done in ${timestamp() - start} ms`);
-        return result;
+        return wrapMultiMap(result, /*isFromPreviousProjectVersion*/ false);
 
-        function key(name: string, alias: Symbol, moduleSymbol: Symbol, checker: TypeChecker) {
+        function key(importedName: string, alias: Symbol, moduleSymbol: Symbol, checker: TypeChecker) {
             const moduleName = stripQuotes(moduleSymbol.name);
             const moduleKey = isExternalModuleNameRelative(moduleName) ? "/" : moduleName;
-            return `${name}|${getSymbolId(skipAlias(alias, checker))}|${moduleKey}`;
+            const original = skipAlias(alias, checker);
+            return `${importedName}|${getSymbolId(original.declarations?.[0].symbol || original)}|${moduleKey}`;
+        }
+
+        function parseKey(key: string) {
+            const symbolName = key.substring(0, key.indexOf("|"));
+            const moduleKey = key.substring(key.lastIndexOf("|") + 1);
+            const ambientModuleName = moduleKey === "/" ? undefined : moduleKey;
+            return { symbolName, ambientModuleName };
+        }
+
+        function wrapMultiMap(map: MultiMap<string, SymbolExportInfo>, isFromPreviousProjectVersion: boolean): SymbolToExportInfoMap {
+            const wrapped: SymbolToExportInfoMap = {
+                get: (importedName, symbol, moduleSymbol, checker) => {
+                    const info = map.get(key(importedName, symbol, moduleSymbol, checker));
+                    return isFromPreviousProjectVersion ? info?.map(info => replaceTransientSymbols(info, checker)) : info;
+                },
+                forEach: (getChecker, action) => {
+                    map.forEach((info, key) => {
+                        const { symbolName, ambientModuleName } = parseKey(key);
+                        action(
+                            isFromPreviousProjectVersion ? info.map(i => replaceTransientSymbols(i, getChecker(i.isFromPackageJson))) : info,
+                            symbolName,
+                            !!ambientModuleName);
+                    });
+                },
+            };
+            if (Debug.isDebugging) {
+                Object.defineProperty(wrapped, "__cache", { get: () => map });
+            }
+            return wrapped;
+
+            function replaceTransientSymbols(info: SymbolExportInfo, checker: TypeChecker) {

```

Comment:
Transient symbols should never have been in this cache, but this isn’t new, it’s just moved and renamed. I’m working on a follow-up to restructure this cache to be safer and less kludgy.